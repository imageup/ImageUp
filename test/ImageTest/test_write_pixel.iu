function main() -> int
{
        image: a = read("./images/google-chrome.jpg");
        
        int : i;
        tuple: x;
        tuple: y;
        float: diff_b;
        float: diff_g;
        float: diff_r;
        for (i=0; i< 50;i=i+1){
        	x = get_pixel(a,(i, i+1));
        	write_pixel(a, (i, i+1), (124,163,45));
        	y = get_pixel(a,(i, i+1));
        	diff_b = y[0]-x[0];
        	diff_g = y[1]-x[1];
        	diff_r = y[2]-x[2];
        	if (diff_b != 0.0 && diff_g != 0.0 && diff_r != 0.0){
        		prints("Done writing pixel");
        	}

        }
		save("test.out.jpg", a);
        return 0;
}
