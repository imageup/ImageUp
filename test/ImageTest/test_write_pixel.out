Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
Done writing pixel
